app/components/Analyst/CBC/CbcEditProjectNumber.tsx

```diff
@@ -0,0 +1,229 @@
+import { useState } from 'react';
+import { graphql, useFragment } from 'react-relay';
+import { useUpdateCbcProjectNumberMutation } from 'schema/mutations/cbc/updateCbcProjectNumber';
+import styled from 'styled-components';
+import { Input } from '@button-inc/bcgov-theme';
+import CircularProgress from '@mui/material/CircularProgress';
+
+interface Props {
+  cbc: any;
+  value: number;
+  isHeaderEditable: boolean;
+}
+
+const StyledH2 = styled.h2`
+  margin: 0;
+  font-size: 16px;
+  cursor: pointer;
+`;
+
+const StyledInput = styled(Input)`
+  & input {
+    margin-top: 8px;
+    margin-bottom: 4px;
+    width: 90%;
+  }
+  & input:focus {
+    outline: ${(props) =>
+      props.error ? '4px solid #E71F1F' : '4px solid #3B99FC'};
+  }
+`;
+
+const StyledDiv = styled('div')`
+  margin-bottom: 8px;
+  display: flex;
+  flex-wrap: wrap;
+`;
+
+const StyledLoadingDiv = styled('div')`
+  margin-top: 8px;
+  margin-bottom: 8px;
+`;
+
+const StyledError = styled('div')`
+  color: #e71f1f;
+  display: inline-block;
+  margin-top: 8px;
+  flex-basis: 100%;
+`;
+
+const StyledMessage = styled('div')`
+  display: flex;
+  &::after {
+    content: '.';
+    visibility: hidden;
+  }
+  flex-basis: 100%;
+`;
+
+const CbcEditProjectNumber: React.FC<Props> = ({
+  cbc,
+  value,
+  isHeaderEditable,
+}) => {
+  const queryFragment = useFragment(
+    graphql`
+      fragment CbcEditProjectNumber_query on Cbc {
+        projectNumber
+      }
+    `,
+    cbc
+  );
+
+  const [baseProjectNumber, setBaseProjectNumber] = useState(value);
+  const [isEditing, setIsEditing] = useState(false);
+  const [isValidating, setIsValidating] = useState(false);
+  const [isSaving, setIsSaving] = useState(false);
+  const [updateProjectNumber] = useUpdateCbcProjectNumberMutation();
+
+  const [projectNumber, setProjectNumber] = useState(
+    queryFragment?.projectNumber
+  );
+
+  const [error, setError] = useState(null);
+
+  const handleSubmit = () => {
```

Comment:
Noticed that after update the project number in dashboard it still shows the old project number.